Sources/PT.PM.Common/UstConstantFolder.cs

```diff
@@ -0,0 +1,281 @@
+                    return null;
+            }
+        }
+
+        private FoldResult TryFoldArrayCreationExpression(ArrayCreationExpression arrayCreationExpression)
+        {
+            if (arrayCreationExpression.Type?.TypeText == "char" &&
+               (arrayCreationExpression.Initializers?.All(i => i is StringLiteral) ?? false))
+            {
+                var value = new StringBuilder();
+                var textSpans = new List<TextSpan>(arrayCreationExpression.Initializers.Count + 1);
+                
+                textSpans.Add(arrayCreationExpression.Type.TextSpan);
+                
+                foreach (StringLiteral stringLiteral in arrayCreationExpression.Initializers)
+                {
+                    value.Append(stringLiteral.Text);
+                    textSpans.Add(stringLiteral.TextSpan);
+                }
+                
+                return new FoldResult(value.ToString(), textSpans);
+            }
+
+            return null;
+        }
+        
+        private FoldResult TryFoldBinaryOperatorExpression(BinaryOperatorExpression binaryOperatorExpression)
+        {
+            FoldResult leftFold = TryFold(binaryOperatorExpression.Left);
+            FoldResult rightFold = TryFold(binaryOperatorExpression.Right);
+            object leftValue = leftFold?.Value;
+            object rightValue = rightFold?.Value;
+            BinaryOperatorLiteral op = binaryOperatorExpression.Operator;
+
+            if ((leftValue is string || leftValue is StringBuilder) && rightFold != null)
+            {
+                if (op.BinaryOperator == BinaryOperator.Plus || op.BinaryOperator == BinaryOperator.Concat)
+                {
+                    // Use StringBuilder instead of immutable strings and concatenation
+                    string rightString = rightValue.ToString();
+
+                    StringBuilder leftStringBuilder;
+                    if (leftValue is string leftString)
+                    {
+                        leftStringBuilder = new StringBuilder((leftString.Length + rightString.Length) * 2);
+                        leftStringBuilder.Append(leftString);
+                    }
+                    else
+                    {
+                        leftStringBuilder = (StringBuilder) leftValue;
+                    }
+                    leftStringBuilder.Append(rightString);
+                    
+                    FoldResult result = ProcessBinaryExpression(binaryOperatorExpression, leftFold, rightFold, leftStringBuilder);
+
+                    if (Logger.IsLogDebugs)
+                    {
+                        Logger.LogDebug(
+                            $"Strings {binaryOperatorExpression} concatenated to \"{leftStringBuilder}\" at {binaryOperatorExpression.TextSpan}");
+                    }
+                    
+                    return result;
+                }
+            }
+            else if (leftValue is long leftInt && rightValue is long rightInt)
```

Comment:
Will floating point values be hadled in future?

Reply:
I think yes but it's not so important for now.